### PR TITLE
Fix ru gen

### PR DESCRIPTION
**Pull request by mattloose**
This code updates unblock all to make it work with batches of reads. It also updates ru_gen to use read batches. This is done by changing the logic of the decision dict to add objects to a list - either for unblock or for stop receiving. At the end of each loop through the code, lists are checked to see if they have actions and - if they do - the batch unblock or batch stop receiving commands are used.

### CODE REVIEW

**Comment by mattloose** [comment]
For info - output of a test run with 0.4 second chunk size but had to use fast and no adapt.

`readfish summary testing.toml /data/Testing/chr20_21/20201004_1732_X2_FAL80952_a9af94fc/
Using reference: /data/references/hg38_simple.mmi
contig  number      sum  min     max    std   mean  median    N50
  chr1    2815  6937124  189  305952  13655   2464     648  57752
 chr10    1213  4282599  264  285789  17434   3531     656  73303
 chr11    1286  4004818  214  297408  15899   3114     650  64584
 chr12    1806  3633948  161  168656   9121   2012     662  29636
 chr13     779  2675870  229  196520  18075   3435     638  94126
 chr14    1008  3717128  232  248134  17031   3688     666  79415
 chr15    1930  4363090  194  150752  11348   2261     639  53082
 chr16    1026  2314660  233  238936  11622   2256     682  45428
 chr17     930  2885510  264  380642  19456   3103     664  95342
 chr18     944  3406665  226  248283  17443   3609     640  85913
 chr19    1014  2359671  205  203583  12726   2327     656  49899
  chr2    3261  6792481  211  274699  11988   2083     642  40291
 chr20     603  2014934  225  210484  16186   3342     668  64090
 chr21      52  1716587  308  332754  60328  33011    9355  97024
 chr22      69   795408  259   77474  14921  11528    5466  21887
  chr3    2413  6557709  189  315418  16275   2718     642  66870
  chr4    2372  7861493  211  310157  17557   3314     666  68608
  chr5    1941  6147170  195  360856  17011   3167     647  82521
  chr6    1946  5178537  213  250400  13877   2661     645  58672
  chr7    1580  4826718  182  457411  18742   3055     651  72464
  chr8    1539  4223112  225  366386  15725   2744     665  53797
  chr9    1764  4823678  219  464521  18246   2735     647  72955
  chrM     220   269807  159   20658   2246   1226     690   1068
  chrX    1630  4807004  213  309386  15830   2949     663  69094
  chrY      12    51223  518   31673   9128   4269     704  31673`

File: ru/ru_gen.py [hunk 1]
```diff
@@ -378,6 +390,14 @@ def simple_analysis(
 
             log_decision()
 
+
+        if len(unblock_batch_action_list)>0:
```
Comment (alexomics) [inline ru/ru_gen.py:394]: We shouldn't need to check the length of these lists sending an empty list should be fine.

**Review by alexomics — CHANGES_REQUESTED** [review]
We've lost the ability to log unblocked read ids. We can fix this by overwriting the `client.unblock_read_batch` function and taking an expanded tuple eg `(channel, read_number, read_id)`. 

We should delete lines rather than commenting them.

**Comment by mattloose** [comment]
Hi @alexomics

This should now be resolved:

```Using reference: /data/references/hg38_simple.mmi
contig  number       sum  min     max    std   mean  median     N50
  chr1    4422   9619928  199  334761  12980   2175     647   53733
 chr10    2341   5974998  244  257624  13396   2552     671   67981
 chr11    2466   5917567  199  304265  13777   2400     660   57902
 chr12    2602   5637225  190  189988  10283   2166     663   44929
 chr13    1191   3749682  229  312928  17908   3148     646  100918
 chr14    1739   4423324  175  246304  13180   2544     663   52295
 chr15    2941   6415141  182  390759  12992   2181     638   52305
 chr16    1225   3162562  233  443993  17741   2582     656   67026
 chr17    1489   4141228  222  335358  17018   2781     666   97667
 chr18    1809   5131435  226  282996  16104   2837     657   71583
 chr19    1498   3179890  202  226417  11809   2123     664   47194
  chr2    5044  11519181  148  431239  13856   2284     649   53720
 chr20    1405   2473512  136  200739  10195   1761     660   29882
 chr21      76   2171861  308  332690  52554  28577    8126   87040
 chr22      89   1560618  259  194784  31744  17535    6729   48197
  chr3    3655   9426730  194  323010  15335   2579     656   67864
  chr4    3910   9942975  194  299593  14177   2543     658   62566
  chr5    3460   8546423  221  261488  13466   2470     652   62021
  chr6    2934   7263398  203  282558  13812   2476     647   68485
  chr7    2969   7504891  177  390557  16347   2528     654   75104
  chr8    2461   5684101  188  350553  14033   2310     651   55010
chr9    2927   7512295  195  428703  16345   2567     660   66258
  chrM     286    325829  251   13357   1720   1139     696    1028
  chrX    2657   6864109  205  294641  14827   2583     666   61302
  chrY      18    457369  394  393336  92139  25409     929  393336```

**Review by alexomics — APPROVED** [review]

**Pull request merged**